api/server/router/local/image.go

```diff
@@ -373,3 +381,16 @@ func (s *router) getImagesSearch(ctx context.Context, w http.ResponseWriter, r *
 	}
 	return httputils.WriteJSON(w, http.StatusOK, query.Results)
 }
+
+func isAuthorizedError(err error) bool {
+	if fError, ok := distribution.IsFallbackError(err); ok {
+		err = fError
+	}
```

Comment:
This doesn't make sense to me. `Pull` should never return a `fallbackError`: https://github.com/docker/docker/blob/master/distribution/pull.go#L136

And I thought we agreed in https://github.com/docker/docker/pull/19756 that the case where fallback is involved is tangential to the regression this PR is trying to fix?